pymeasure/instruments/redpitaya/redpitaya_scpi.py

```diff
@@ -0,0 +1,338 @@
+        values={True: 1, False: 0},
+    )
+
+
+class DigitalChannelLed(Channel):
+    """ A LED digital line (Output only)"""
+
+    enabled = Channel.control(
+        "DIG:PIN? LED{ch}", "DIG:PIN LED{ch},%d",
+        """ Control the enabled state of the led (bool)""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, False: 0},
+    )
+
+
+class AnalogInputSlowChannel(Channel):
+    """ A slow analog input channel"""
+
+    voltage = Channel.measurement(
+        "ANALOG:PIN? AIN{ch}",
+        """ Measure the voltage on the corresponding analog input channel, range is [0, 3.3]V""",
+    )
+
+
+class AnalogOutputSlowChannel(Channel):
+    """ A slow analog output channel"""
+
+    voltage = Channel.setting(
+        "ANALOG:PIN AOUT{ch}, %f",
+        """ Set the voltage on the corresponding analog input channel, range is [0, 1.8]V""",
+        validator=truncated_range,
+        values=[0, 1.8],
+    )
+
+
+class AnalogInputFastChannel(Channel):
+
+    gain = Instrument.control(
+        "ACQ:SOUR{ch}:GAIN?", "ACQ:SOUR{ch}:GAIN %s",
+        """Control the gain of the selected fast analog input either 'LV' or 'HV'
+        (see jumpers on boards)
+
+        'LV' set the returned values in the range [-1, 1]V and 'HV' in the range [-20, 20]V
+        """,
+        validator=strict_discrete_set,
+        values=['LV', 'HV'],
+    )
+
+    def get_data_from_ascii(self) -> np.ndarray:
+        self.write("ACQ:SOUR{ch}:DATA?")
+        data_str = self.read()
+        return np.fromstring(data_str.strip('{}').encode(), sep=',')
+
+    def get_data_from_binary(self) -> np.ndarray:
+        """ Get the data in """
+        self.write("ACQ:SOUR{ch}:DATA?")
+        self.read_bytes(1)
+        nint = int(self.read_bytes(1).decode())
+        length = int(self.read_bytes(nint).decode())
+        data = np.frombuffer(self.read_bytes(length), dtype=int)
+        self.read_bytes(2)
+        if self.gain == 'LV':
+            max_range = 2 * RedPitayaScpi.LV_MAX
+        else:
+            max_range = 2 * RedPitayaScpi.HV_MAX
+
+        return max_range * data / (2**16 - 1) - max_range / 2
+
+
+class RedPitayaScpi(Instrument):
+    """This is the class for the Redpitaya reconfigurable board
+
+    The instrument is accessed using a TCP/IP Socket communication, that is an adapter in the form:
+    "TCPIP::x.y.z.k::port::SOCKET" where x.y.z.k is the IP address of the SCPI server
+    (that should be activated on the board) and port is the TCP/IP port number, usually 5000
+
+    To activate the SCPI server, you have to connect first the redpitaya to your computer/network
+    and enter the url address written on the network plug (on the redpitaya). It should be something
+    like "RP-F06432.LOCAL/" then browse the menu, open the Development application and activate the
+    SCPI server. When activating the server, you'll be notified with the IP/port address to use
+    with this Instrument.
+    """
```

Comment:
It is good to mention the creation of the address in the instrument's docstring:
```suggestion

    :param ip_address: IP address to use, if `adapter` is None.
    :param port: Port number to use, if `adapter` is None.
    """
```